### PR TITLE
Add regex for non-update rpm and replace assert with die command

### PR DESCRIPTION
**Pull request by b10n1k**
Apply feedback from https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11691#discussion_r552483085
In addition, add regex to match Version differences when the report does not
provide any update.

- Related ticket: https://progress.opensuse.org/issues/71050
- Verification run: 
- [no updates](http://aquarius.suse.cz/tests/4343#details)
- [with updates](http://aquarius.suse.cz/tests/4342)

### CODE REVIEW

File: lib/containers/container_images.pm [hunk 1]
```diff
@@ -156,20 +155,24 @@ sub test_opensuse_based_image {
...
+            last;
+        }
         if ($line =~ $regex2match) {
             # Use of Dotted-Decimal-Versions. Do not remove 'v' prefix
             my $updated_v = version->parse("v$+{update_version}");
             my $stable_v  = version->parse("v$+{stable_version}");
             record_info("checking... $+{package}", "$+{package} $stable_v to $updated_v");
-            assert_true($updated_v > $stable_v, "$+{package} $stable_v is not updated to $updated_v");
+            die "$+{package} $stable_v is not updated to $updated_v" unless ($updated_v > $stable_v);
         }
```
Comment (pdostal) [inline lib/containers/container_images.pm:174]: What about `else { … }` statement? It might be useful for incidental debugging.

**Review by pdostal — APPROVED** [review]
I agree 🚀

File: lib/containers/container_images.pm (hunk 1 above)
Comment (b10n1k) [reply lib/containers/container_images.pm:174]: else what... not sure. i assume that you talk about the $line matching. thus if the regex is correct will match only the ones that have the packages. i do not think we care for all the other $lines atm.

Comment (grisu48) [reply lib/containers/container_images.pm:174]: I think Pavel means the following:

        if ($line =~ $regex2zerorpm) {
        ...
        } elsif ($line =~ $regex2match) {
        ...
        } else {
            // we don't care about the other lines for now
        }

And to add a check, if at least one of the checks is happening.

**Review by grisu48 — APPROVED** [review]

File: lib/containers/container_images.pm (hunk 1 above)
Comment (b10n1k) [reply lib/containers/container_images.pm:174]: xm. i believe it is the same. but now you make think which version is more computational efficient. let me try.

Comment (b10n1k) [reply lib/containers/container_images.pm:170]: if..else seems faster than multi_ifs. and counting also the prediction factor i ended up with switching the blocks. i hope it looks better now(reopen it if you feel that something is still missing). thanks for the inspiration.

**Review by pdostal — APPROVED** [review]

**Review by grisu48 — APPROVED** [review]

**Pull request merged**